firebase.go

```diff
@@ -48,6 +48,9 @@ type Firebase struct {
 	params _url.Values
 	client *http.Client
 
+	eventMtx   sync.Mutex
+	eventFuncs map[string]chan struct{}
```

Comment:
I wouldn't necessarily call these semaphores. This follows the same pattern as the `watch` function which is referred to as `stopWatching` - however `eventStoppers` sounds odd, funny, but odd.